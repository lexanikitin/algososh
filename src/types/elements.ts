import {ElementStates} from "./element-states";

export type TElementStr = { value: string, state: ElementStates };
export type TElementNum = { value: number, state: ElementStates };
