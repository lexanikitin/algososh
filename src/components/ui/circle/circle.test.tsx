import {Circle} from "./circle";
import renderer from "react-test-renderer";
import {ElementStates} from "../../../types/element-states";

describe('Отрисовка компонента Circle:', () => {
  it('без буквы', () => {
    const testTree = renderer
      .create(<Circle/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с буквами', () => {
    const testTree = renderer
      .create(<Circle letter={'test'}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с head', () => {
    const testTree = renderer
      .create(<Circle head={'test'}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с react-элементом в head', () => {
    const testTree = renderer
      .create(<Circle head={<Circle/>}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с tail', () => {
    const testTree = renderer
      .create(<Circle tail={'test'}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с react-элементом в tail', () => {
    const testTree = renderer
      .create(<Circle tail={<Circle/>}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с index', () => {
    const testTree = renderer
      .create(<Circle index={1}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('с пропом isSmall ===  true', () => {
    const testTree = renderer
      .create(<Circle isSmall={true}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('в состоянии Default', () => {
    const testTree = renderer
      .create(<Circle state={ElementStates.Default}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('в состоянии Changing', () => {
    const testTree = renderer
      .create(<Circle state={ElementStates.Changing}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
  it('в состоянии Modified', () => {
    const testTree = renderer
      .create(<Circle state={ElementStates.Modified}/>)
      .toJSON();
    expect(testTree).toMatchSnapshot();
  });
});
