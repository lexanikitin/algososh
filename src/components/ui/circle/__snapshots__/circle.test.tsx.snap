// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Отрисовка компонента Circle: без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: в состоянии Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: в состоянии Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: в состоянии Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Отрисовка компонента Circle: с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`Отрисовка компонента Circle: с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Отрисовка компонента Circle: с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
